src/Riok.Mapperly/Emit/SourceEmitter.cs

```diff
@@ -8,16 +8,19 @@ namespace Riok.Mapperly.Emit;
 
 public static class SourceEmitter
 {
-    public static CompilationUnitSyntax Build(MapperDescriptor descriptor)
+    public static MemberDeclarationSyntax BuildMember(MapperDescriptor descriptor)
     {
         var sourceEmitterContext = new SourceEmitterContext(descriptor.Symbol.IsStatic, descriptor.NameBuilder);
         MemberDeclarationSyntax member = ClassDeclaration(descriptor.Syntax.Identifier)
             .WithModifiers(descriptor.Syntax.Modifiers)
             .WithMembers(List(BuildMembers(descriptor, sourceEmitterContext)));
 
         member = WrapInClassesAsNeeded(descriptor.Symbol, member);
-        member = WrapInNamespaceIfNeeded(descriptor.Namespace, member);
+        return WrapInNamespaceIfNeeded(descriptor.Namespace, member);
+    }
```

Comment:
Couldn't the `CompilationUnitSyntax` be returned here, just without the `NormalizeWhiteSpace`? Then the MapperGenerator could call `NormalizeWhiteSpace` in the very last step.